backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/tutor/submission/RemoveSubmittedQuestionTest.groovy

```diff
@@ -0,0 +1,148 @@
+package pt.ulisboa.tecnico.socialsoftware.tutor.submission.service
+
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.boot.test.context.TestConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.SpockTest
+import pt.ulisboa.tecnico.socialsoftware.tutor.BeanConfiguration
+import pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.ErrorMessage
+import pt.ulisboa.tecnico.socialsoftware.tutor.exceptions.TutorException
+import pt.ulisboa.tecnico.socialsoftware.tutor.question.domain.Question
+import pt.ulisboa.tecnico.socialsoftware.tutor.submission.domain.Review
+import pt.ulisboa.tecnico.socialsoftware.tutor.submission.domain.Submission
+import pt.ulisboa.tecnico.socialsoftware.tutor.user.User
+
+
+@DataJpaTest
+class RemoveSubmittedQuestionTest extends SpockTest{
+    def student
+    def teacher
+    def question
+    def submission
+
+    def setup() {
+        student = new User(USER_1_NAME, USER_1_USERNAME, User.Role.STUDENT)
+        student.setEnrolledCoursesAcronyms(courseExecution.getAcronym())
+        userRepository.save(student)
+        teacher = new User(USER_2_NAME, USER_2_USERNAME, User.Role.TEACHER)
+        userRepository.save(teacher)
+        question = new Question()
+        question.setKey(1)
+        question.setTitle(QUESTION_1_TITLE)
+        question.setContent(QUESTION_1_CONTENT)
+        question.setCourse(course)
+        question.setStatus(Question.Status.IN_REVIEW)
+        questionRepository.save(question)
+        submission = new Submission()
+        submission.setQuestion(question)
+        submission.setUser(student)
+        submission.setCourseExecution(courseExecution)
+        submissionRepository.save(submission)
+    }
+
+    def "teacher removes a submitted question with an associated review"(){
+        given: "a review"
+        def review = new Review()
+        review.setComment(REVIEW_1_COMMENT)
+        review.setUser(teacher)
+        review.setSubmission(submission)
+        review.setStatus("IN_REVISION")
+        submission.addReview(review)
+        reviewRepository.save(review)
+
+        when:
+        questionService.removeQuestion(teacher.getId(), question.getId())
```

Comment:
we have to discuss this interface, important to distinguish removeQuestion from removeQuestionSubmission